### DISCUSSION TITLE
Maui/XAML: Specific Navigation Pattern W/ Login page (no back) and Shell/NavigationPage

### DISCUSSION THREAD

**Discussion by stevencreaney** [dotnet/maui#17590] [Q&A]
I need some assistance, mainly with a good pattern for accomplishing this specific pattern of navigation.
-Login
-Home
	-Appointments
		-Details (by appointment)
			-...many subviews for business functions
	-Notifications
	-Reports
		-Report (by id)

What I'm struggling with is how to accomplish the Login page to Home.  The login page MUST be shown right away (users cannot use the app at all unless authenticated using MSAL).  Once logged in, then navigate to the Home page.  The user cannot go back to the Login page, I do not want to show the back button.  

I've tried doing NavigationPages, but I don't see how to do Deferrals when navigating, so loading/saving data is difficult without completely rolling my own navigation.  

I've tried using a Shell, but I can't find how to properly 'hide' the Login page from the shell navigation. 

I've also tried loading right to the login page, then replacing MainPage with my Shell, but I just get 'No installed components were detected.'

What is a good practice/pattern for doing something like this?  Frame navigation in UWP was super easy, however, this really seems to be difficult to get into complex business scenarios.

**Comment by robintschroeder** [comment]
There are basically three ways to do navigation.
1. Shell
2. Simple MVVM page navigation - for this, you just use the built in [navigation stack](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/pages/navigationpage#manipulate-the-navigation-stack)
3. MVVM pagemodel navigation - this requires that you use DI or at least an IoC container, you maintain the different navigation stacks with some sort of a singleton service, reassigning the Application.MainPage. This is basically a homegrown version of Shell, but you would be able to build out and maintain a variety of navigation stacks, either single stacks, or a tab stack, etc. If you don't want to build this yourself, you could take a look at [FreshMVVM](https://michaelridland.com/freshmvvm-maui/a-first-look-with-freshmvvm-maui/), Prism or  MVVMCross. There are also options like MVVM Toolkit.
There are lots of people on the interwebs contemplating [this very thing](https://www.reddit.com/r/dotnetMAUI/comments/12arji3/mvvm_frameworks_for_maui/).
Best of Luck!
